pkg/endpoint/endpoint.go

```diff
@@ -2389,12 +2389,8 @@ func (e *Endpoint) identityResolutionIsObsolete(myChangeRev int) bool {
 
 // Must be called with e.Mutex NOT held.
 func (e *Endpoint) runLabelsResolver(owner Owner, myChangeRev int) {
-	// NOTE: UnconditionalLock is used here only for logging
-	e.UnconditionalLock()
```

Comment:
I'm on the fence about this one because `GetOpLabels` is more costly than doing `e.OpLabels.IdentityLabels()`. Can't you replace the Lock with RLock for this case?

Reply:
Sure.

Reply:
For reference, what are you concerned is expensive? Iterating through the `Labels` structure to convert it into a string array?

Reply:
Updated, please revise.

Reply:
Yes.

Looks super good now